ArcGISRuntimeSDKQt_CppSamples/CloudAndPortal/OAuthRedirectExample/README.md

```diff
@@ -0,0 +1,121 @@
+
+Next time you click a link of protocol-type foo you will be asked which application to open it with - enter the path to your app. You can try out the protocol by entering `myapp://hello` in a new Firefox tab. 
+
+
+When the app is launched from the browser a new instance will be launched: however, in this example,
+the desired behaviour is that the same instance of the app should be notified with the authorization code. To achieve this, on linux the sample makes use of the `QtSingleApplication` from qt-solutions. Using this type, the original instance is notified of the authorization code as a command line argument.
+
+### Mac
+In order to register a custom URL protocol on Mac whilst developing, you can define a scheme in XCode.
+
+First, build the project in QtCreator. Next, Tt create an XCode project from QtCreator add the follwoing additional arguments add additional arguments to the project's qmake build settings
+
+    `-spec macx-xcode`
+
+Open the generated project (e.g. `OAuthRedirectExample.xcodeproj`) in XCode. 
+
+You may need to adjust the Qt Postlink step created by QtCreator to point to the target which XCode will build (e.g. to reference the OAuthRedirectExample.app in the sub directory for Debug or Release). E.g. Go to Build Phases/Qt PostLink and edit the script to look something like:
+
+`
+install_name_tool -change libEsriCommonQt.dylib "~/ArcGIS_SDKs/Qt100.1/sdk/macOS/x64/lib/libEsriCommonQt.dylib" Release/OAuthRedirectExample.app/Contents/MacOS/OAuthRedirectExample
+`
+
+Next go to Info/URL Types and add a new URL type with Identifier and URL Schemes set to the custom URL scheme you set up for your app.
+
+Now when you build and run the Project from XCode, the url scheme will be recognised and the URL will be treated as a `QFileOpenEvent`.
+
+_NOTE: When deploying an App on mac it should be possible to set the URL type in the Info.Plist as is done for iOS platforms._
+
+### Android
+On Android, the custom URL scheme is handled with an Activity (see `MyActivity.java`) that handles an intent to open the custom Url by calling a JNI function exposed by the application. 
+
+The Custom url scheme is set up in the `AndroidManifest.xml` file. You need to update `<data android:scheme="exampleapp"/>` to use your own scheme in place of "exampleapp".
+
+### iOS
+On iOS, the `Info.plist` file defines the custom url scheme. You need to update the following code with your custome URL scheme in place of `exampleapp`:
+
+`
+<key>CFBundleURLSchemes</key>
+<array>
+    <string>exampleapp</string>
+</array>
+`
+
+## QtSingleApplication dependancy
+On Windows and Linux, the sample depends upon the `QtSingleApplication` type from qt-solutions. You can download the project from 
+https://github.com/qtproject/qt-solutions. The sample is designed to include the source files ditrectly: update the `QT_SINGLEAPP_DIR` 
```

Comment:
update this to use url redirect - `http://links.esri.com/qtSingleApplication`. Perhaps it could say "... project from [GitHub](http://links.esri.com/qtSingleApplication)"